internal/httpproxy/http_proxy.go

```diff
@@ -176,20 +178,22 @@ func (p *ProxyConn) authenticate() error {
 
 	// Parse and validate HTTP request, expecting CONNECT with
 	// valid token and signature
-	response := httpResponseString(http.StatusOK)
+	httpCode := http.StatusOK
 
 	connectInfo, err := p.ConnectValidator.ParseConnect(req, ekm)
 	if err != nil {
 		var httpErr *connect.HTTPError
 		if errors.As(err, &httpErr) {
-			response = httpResponseString(httpErr.Code)
+			httpCode = httpErr.Code
 		} else {
 			p.logger.Error("failed to parse CONNECT:", zap.Error(err))
 
-			response = httpResponseString(http.StatusBadRequest)
+			httpCode = http.StatusBadRequest
 		}
 	}
 
+	response := httpResponseString(httpCode)
```

Comment:
nit: I would move this to line 207 so the variable is declared nearer to its 1st usage.